project/CodeStone1125/README.md

```diff
@@ -0,0 +1,182 @@
+# 2D Collision Detector
+
+
+## Basic Information
+
+Simulate how different type of shape collide in a 2D map.
+
+* github repository: [https://github.com/CodeStone1125/2DCollisionDetector](https://github.com/CodeStone1125/2DCollisionDetector)
+
+## Problem to Solve
+
+The objective of this system is to calculate scenarios for the most common shapes in
+a 2D map, such as rectangles, circles, and polygons. Before we dive deeper into the 
+topic, let's first discuss the "Axis-Aligned Bounding Box" (AABB) algorithm. While 
+it's not a single algorithm but a framework for collision detection, it forms the basis
+for many collision detection systems in 2D graphics and game development due to its 
+simplicity and efficiency. Moreover, in the 2D world, AABB can effectively handle 
+approximate collision detection for most cases.
+
+However, for more complex polygons, we need to introduce another algorithm called the
+Separating Axis Theorem (SAT). SAT is more precise but also more resource-intensive.
+For most cases in the 2D world, we canhandle collision detection through the following
+steps:
+
+1. Preliminary screening of collision objects using AABB.
+2. Further checking with SAT.
+
+
+### AABB's Pseudo code
+* Bounding Boxes: Each object in a 2D scene is enclosed by a rectangle (a bounding box)
+   that is aligned with the axes of the coordinate system.These rectangles are often
+   referred to as "AABBs" because they are aligned with the axes.
+   Here is the pseudo code in `python`:
+  
+  ```python
+  def calculate_aabb(points):
+    if len(points) == 0:
+        return None
+
+    # Initialize min and max values with the first point
+    min_x, min_y = points[0]
+    max_x, max_y = points[0]
+
+    # Iterate through the points to find the AABB
+    for x, y in points:
+        if x < min_x:
+            min_x = x
+        elif x > max_x:
+            max_x = x
+
+        if y < min_y:
+            min_y = y
+        elif y > max_y:
+            max_y = y
+
+    # Return the AABB as a tuple of (min_x, min_y, max_x, max_y)
+    return (min_x, min_y, max_x, max_y)
+  ```
+### Example usage:
+
+points = [(1, 2), (3, 4), (-1, 0), (5, 6)]
+
+aabb = calculate_aabb(points)
+
+print(aabb)  # Output will be the AABB: (-1, 0, 5, 6)
+### SAT's Pseudo code
+* Separating Axis Theorem:  A collision detection method for 2D shapes. It checks if
+  there's an axis where two shapes can be separated, indicating no collision.
+  If no separating axis is found, the shapes are colliding.  SAT is used for accurate
+  collision detection in games and simulations.
+  Here is the pseudo code in `python`:
+  
+  ```python
```

Comment:
I like the way you use Python for showing the algorithms and ideas.  It is a very good way for coding discussions.